index.bs

```diff
@@ -225,7 +225,17 @@ its \[[Prototype]] an empty object, created as if by ObjectCreate(%ObjectPrototy
 
 <h4 id="assert" oldids="assert-condition-data,dom-console-assert" method for="console">assert(<var>condition</var>, ...<var>data</var>)</h4>
 
-If <var>condition</var> is false, perform Logger("error", <var>data</var>).
+<emu-alg>
+  1. If _condition_ is true, abort these steps.
+  1. Let _message_ be a string without any formatting specifiers indicating generically an assertion failure (such as "Assertion failed").
+  1. If _data_ is empty, append _message_ to _data_.
+  1. Otherwise, implementations should perform these substeps:
+    1. Let _first_ be the first element of _data_.
+    1. If _first_ is not a string, prepend _message_ to _data_. Abort these substeps.
```

Comment:
The intention was a type test, mainly because `concat` is not well-defined for non-string types. An alternative would have been to do the following in the next line:

> Let _concat_ be the concatenation of _message_, U+003A COLON (:), U+0020 (SPACE), and [toString](https://tc39.github.io/ecma262/#sec-tostring)(_first_).

However that would lead to the following behavior, which is unintended and also would not match any current behavior:

```
console.assert(false, { foo: bar }) // "Assertion failed: [Object object]"
```

Will push a commit with the proper type test notation.